### PR TITLE
Add CDC polling mode implementation

### PR DESCRIPTION
**Pull request by chathuranga95**
## Automation tests
 - Integration tests
   > Please find the integration test running instructions from README.md file.

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? 
> Yes
 - Ran FindSecurityBugs plugin and verified report?
> Yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? 
> Yes

## Samples
```
@source(type = 'cdc', mode='polling', polling.column = 'id', jdbc.driver.name = 'com.mysql.jdbc.Driver',
url = 'jdbc:mysql://localhost:3306/SimpleDB', username = 'cdcuser', password = 'pswd4cdc', 
table.name = 'students', 
@map(type='keyvalue'), @attributes(id = 'id', name = 'name'))
define stream inputStream (id int, name string);
```
In this example, the cdc source starts polling students table for inserts. polling.column is an auto incremental field. url, username, password and jdbc.driver.name are used to connect to the database.

```
@source(type = 'cdc', mode='polling', polling.column = 'id', datasource.name = 'SimpleDB',
table.name = 'students', @map(type='keyvalue'), @attributes(id = 'id', name = 'name'))
define stream inputStream (id int, name string);
```
In this example, the cdc source starts polling students table for inserts. polling.column is a char column with the pattern S001, S002, ... . datasource.name is used to connect to the database. Note that the datasource.name works only with the Stream Processor.

```
@source(type = 'cdc', mode='polling', polling.column = 'last_updated',
datasource.name = 'SimpleDB', table.name = 'students', @map(type='keyvalue'))
define stream inputStream (name string);
```
In this example, the cdc source starts polling students table for inserts and updates. polling.column is a timestamp field.

## Test environment
- JDK version: 1.8
- Operating system: Ubuntu 16.04
- Databases: MySQL 5.7.23, H2 1.4.196, Postgres 10.6, Oracle 12.2.0.1-ee, Oracle 11.2.0.2-xe,
  MS-SQL Server 14.0
- Docker version 18.09.0
- Maven version 3.3.9

### CODE REVIEW

File: component/pom.xml [hunk 1]
```diff
@@ -186,9 +214,265 @@
                                 <PORT>${siddhi-cdc-mysql.port}</PORT>
                                 <DOCKER_HOST_IP>${docker.container.siddhi-cdc-mysql.ip}</DOCKER_HOST_IP>
                             </environmentVariables>
-                            <suiteXmlFiles>
-                                <suiteXmlFile>src/test/resources/testng.xml</suiteXmlFile>
-                            </suiteXmlFiles>
+                            <test>
+                                org.wso2.extension.siddhi.io.cdc.source.TestCaseOfMySQLCDC
```
Comment (Niveathika92) [inline component/pom.xml]: Why is there different test cases for each type of database? Isn't it enough to parameterise and reuse the same test cases?

File: component/src/main/resources/query-config.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  - name: "mysql"
+    selectQuery: SELECT {{COLUMN_LIST}} FROM {{TABLE_NAME}} {{CONDITION}}
+  - name: "oracle"
+    selectQuery: SELECT {{COLUMN_LIST}} FROM {{TABLE_NAME}} {{CONDITION}}
+  - name: "PostgreSQL"
+    selectQuery: SELECT {{COLUMN_LIST}} FROM {{TABLE_NAME}} {{CONDITION}}
+  - name: "H2"
+    selectQuery: SELECT {{COLUMN_LIST}} FROM {{TABLE_NAME}} {{CONDITION}}
+  - name: "Microsoft SQL Server"
+    selectQuery: SELECT {{COLUMN_LIST}} FROM {{TABLE_NAME}} {{CONDITION}}
```
Comment (Niveathika92) [inline component/src/main/resources/query-config.yaml]: Add a new line at EOF

File: component/src/test/java/org/wso2/extension/siddhi/io/cdc/source/TestCaseOfH2CDC.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+import org.wso2.siddhi.core.stream.input.InputHandler;
+import org.wso2.siddhi.core.stream.output.StreamCallback;
+import org.wso2.siddhi.core.util.SiddhiTestHelper;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class TestCaseOfH2CDC {
+
+    private static final Logger log = Logger.getLogger(TestCaseOfH2CDC.class);
```
Comment (Niveathika92) [inline component/src/test/java/org/wso2/extension/siddhi/io/cdc/source/TestCaseOfH2CDC.java]: Reuse test classes  as mentioned earlier

File: docs/api/1.0.2.md [hunk 4]
```diff
@@ -4,11 +4,11 @@
 
 ### cdc *<a target="_blank" href="https://wso2.github.io/siddhi/documentation/siddhi-4.0/#source">(Source)</a>*
```
Comment (Niveathika92) [inline docs/api/1.0.2.md]: This is a generated class, do not commit

File: component/src/main/resources/query-config.yaml (hunk 2 above)
Comment (chathuranga95) [reply component/src/main/resources/query-config.yaml]: Fixed with commit: 952809e9657c776e5a8dae55a06bda5c0f46d2dd

File: docs/api/1.0.2.md (hunk 4 above)
Comment (chathuranga95) [reply docs/api/1.0.2.md]: Fixed with commit: dfdf273f8aa49429a4ff79b2ef25af926ff20e13

File: component/src/test/java/org/wso2/extension/siddhi/io/cdc/source/TestCaseOfH2CDC.java (hunk 3 above)
Comment (chathuranga95) [reply component/src/test/java/org/wso2/extension/siddhi/io/cdc/source/TestCaseOfH2CDC.java]: Fixed with commit: 3d57598

File: component/pom.xml (hunk 1 above)
Comment (chathuranga95) [reply component/pom.xml]: Fixed with commit: 3d57598

File: pom.xml [hunk 5]
```diff
@@ -160,10 +195,23 @@
...
         <siddhi-store-rdbms.version>4.0.39</siddhi-store-rdbms.version>
+        <carbon.datasource.core.version>RELEASE</carbon.datasource.core.version>
+        <mavan.findbugsplugin.exclude.file>../findbugs-exclude.xml</mavan.findbugsplugin.exclude.file>
+        <ojdbc6.version>12.1.0.1-atlassian-hosted</ojdbc6.version>
+        <sqljdbc4.version>4.0</sqljdbc4.version>
+        <postgresql.version>RELEASE</postgresql.version>
+        <h2.connector.version>1.4.187</h2.connector.version>
         <siddhi-cdc-mysql.port>3309</siddhi-cdc-mysql.port>
         <docker.container.siddhi-cdc-mysql.ip>0.0.0.0</docker.container.siddhi-cdc-mysql.ip>
+        <siddhi-cdc-oracle.port>1522</siddhi-cdc-oracle.port>
```
Comment (Niveathika92) [inline pom.xml]: If these are related to plugin arguments, then move inside config tag of each plugin call

Comment (chathuranga95) [reply pom.xml]: variables like `siddhi-cdc-oracle.port` are defined in the parent pom because they are being used inside more than one plugin.

**Pull request merged**